search-ui/src/pages/Details/Details.js

```diff
@@ -34,37 +38,116 @@ export default function Details() {
 
   }, [id]);
 
+  useEffect(() => {
+    setHighlight(null);
+  }, [q]);
+
+  function GetTagsHTML(tags) {
+
+    if (!!tags) {
+      let tagsHtml = tags.map((tagValue, index) => {
+        if (index < 10) {
+
+          if (tagValue.length > 30) { // check tag name length
+            // create substring of tag name length if too long
+            tagValue = tagValue.substring(0, 30);
+          }
+
+          return <button key={index} className="tag" onClick={() => setQ(tagValue)}>{tagValue}</button>;
+        } else {
+          return null;
+        }
+      });
+
+      return tagsHtml;
+    }
+
+    return null;
+  }
+
+  let tags = GetTagsHTML(document.keyPhrases);
+
+  function GetSnippets(q, content) {
+    if (!!content && q.trim() !== "") {
+      var regex = new RegExp(q, 'gi');
+
+      let matches = content.match(regex);
+
+      return matches.map((value, index) => {
+        var startIdx;
+        var ln = 400;
```

Comment:
Please add documentation here to clarify these hardcoded values.  
`var ln = 400; // 400 is max length of snippet`

Or even better: 
```
var maxLengthOfSnippet = 400;
var ln = maxLengthOfSnippet;

```
And then use maxLengthOfSnippet / 2 instead of the 200 below if this is just halving it, in case we should want to easily change snippet size later.

Reply:
Great suggestion! I took the latter example